packages/adapter-vercel/index.js

```diff
@@ -467,20 +467,38 @@ function static_vercel_config(builder, config) {
 		overrides[page.file] = { path: overrides_path };
 	}
 
+	const routes = [
+		...prerendered_redirects,
+		{
+			src: `/${builder.getAppPath()}/immutable/.+`,
+			headers: {
+				'cache-control': 'public, immutable, max-age=31536000'
+			}
+		}
+	];
+
+	if (process.env.VERCEL_SKEW_PROTECTION_ENABLED) {
+		routes.push({
+			src: '/.*',
+			has: {
+				type: 'header',
+				key: 'Sec-Fetch-Dest',
+				value: 'document'
+			},
+			headers: {
+				'Set-Cookie': `__vdpl=${process.env.VERCEL_DEPLOYMENT_ID}; Path=/${builder.config.kit.paths.base}; SameSite=Strict; Secure; HttpOnly`
```

Comment:
previously you deleted the cookie for the `version.json` route - is that not necessary after all?

Reply:
correct — we only care about non-document requests for this file. AFAICT there's no way to set multiple cookies (you can't do `set-cookie: [...]`, and if you add a header in multiple `continue` routes the last one wins), so we had to get rid of it

Reply:
wait, no, hang on... this _is_ a problem — the `version.json` will be requested with the old deployment ID. gah

Reply:
(this seemed to work in my testing, because the cookie was still in my browser from the earlier iteration 🤦 )

Reply:
I've 'fixed' this by setting the second `set-cookie` header on the response for the entry point, since we know it will happen before the client hydrates. Since adapters don't have access to the build manifest, this involves some hackery, but hopefully we can remove it in future once the Build Output API supports setting multiple `set-cookie` headers in a single response.